### PR TITLE
Introduce the license maven plugin

### LINKED ISSUE #33: Using the License Maven Plugin
To make sure that all files in the repos have an update-to-date license header, we should use the [`License Maven Plugin`](http://www.mojohaus.org/license-maven-plugin/).

### PR DESCRIPTION
**Pull request by dacrome**
Resolves #33

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 The MIT License (MIT)
 
-Copyright (c) 2015 OSIAM
+Copyright (C) 2016 tarent solutions GmbH
```
Comment (wallner) [inline LICENSE]: Please change it to (c) 2013-2016 tarent solutions GmbH

File: pom.xml [hunk 2]
```diff
@@ -327,6 +328,32 @@
...
+                <artifactId>license-maven-plugin</artifactId>
+                <version>2.11</version>
+                <configuration>
+                    <header>src/main/license/header.txt</header>
+                    <excludes>
+                        <exclude>circle.yml</exclude>
+                        <exclude>jpmInstall.sh</exclude>
+                        <exclude>**/*.md</exclude>
+                        <exclude>bootstrap.min.css</exclude>
+                        <exclude>src/test/**</exclude>
```
Comment (wallner) [inline pom.xml]: Why are you excluding this file?

File: pom.xml [hunk 3]
```diff
@@ -327,6 +328,32 @@
...
+                <groupId>com.mycila</groupId>
+                <artifactId>license-maven-plugin</artifactId>
+                <version>2.11</version>
+                <configuration>
+                    <header>src/main/license/header.txt</header>
+                    <excludes>
+                        <exclude>circle.yml</exclude>
+                        <exclude>jpmInstall.sh</exclude>
+                        <exclude>**/*.md</exclude>
+                        <exclude>bootstrap.min.css</exclude>
```
Comment (wallner) [inline pom.xml:342]: Why are you excluding this file?

File: pom.xml [hunk 4]
```diff
@@ -327,6 +328,32 @@
...
+                        <exclude>src/test/**</exclude>
+                        <exclude>**/LogicalOperatorRules.g4</exclude>
```
Comment (wallner) [inline pom.xml]: Why are you excluding this file?

File: pom.xml [hunk 5]
```diff
@@ -1,27 +1,28 @@
...
-  ~ included in all copies or substantial portions of the Software.
-  ~
-  ~ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
-  ~ EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
-  ~ MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
-  ~ IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
-  ~ CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
-  ~ TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
-  ~ SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
-  -->
```
Comment (wallner) [inline pom.xml:23]: Please add a blank line at the end of the license.

File: LICENSE (hunk 1 above)
Comment (dacrome) [inline LICENSE]: Ok

File: pom.xml (hunk 3 above)
Comment (dacrome) [inline pom.xml:342]: Because it's a copy from http://getbootstrap.com/

File: pom.xml (hunk 2 above)
Comment (dacrome) [inline pom.xml]: Because the files in the test folder are not shipped, right?

File: pom.xml (hunk 4 above)
Comment (dacrome) [inline pom.xml]: the license maven plugin does not know this file extension, but I added a [custom mapping](supported-comment-types)

File: pom.xml (hunk 5 above)
Comment (dacrome) [inline pom.xml:23]: After `-->` in [line 25](https://github.com/osiam/osiam/pull/156/files#diff-600376dffeb79835ede4a0b285078036R25)?

File: pom.xml (hunk 2 above)
Comment (wallner) [inline pom.xml]: Of course they are. They are part of the source code.

File: pom.xml (hunk 3 above)
Comment (wallner) [inline pom.xml:342]: Ah. I see.

File: pom.xml (hunk 5 above)
Comment (wallner) [inline pom.xml:23]: Yes, just swap the end of comment and the last (blank) line of the license.

**Comment by dacrome** [comment]
Updated

File: LICENSE (hunk 1 above)
Comment (dacrome) [inline LICENSE]: Ah, this is the [LICENSE](https://github.com/osiam/osiam/blob/7e95f37dd9869a77e373d5fa800a7260ec4223e5/LICENSE) file. should I remove this file?

Comment (dacrome) [inline LICENSE]: Ok, I use this file as the header for the license plugin

File: LICENSE [hunk 6]
```diff
@@ -18,5 +18,4 @@ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
-SOFTWARE.
-
+SOFTWARE.
```
Comment (wallner) [inline LICENSE]: Please don't remove the last line.

Comment (dacrome) [inline LICENSE]: Done

**Comment by wallner** [comment]
Excellent. Thank you.

**Pull request merged**